### PR TITLE
devops: add bot for Chromium --headless=new

### PR DESCRIPTION
**Pull request by mxschmitt**
https://github.com/microsoft/playwright/issues/23389

### CODE REVIEW

File: .github/workflows/tests_secondary.yml [hunk 1]
```diff
@@ -656,3 +656,24 @@ jobs:
...
+    - uses: actions/setup-node@v3
+      with:
+        node-version: 16
+    - run: npm ci
+      env:
+        DEBUG: pw:install
+        PLAYWRIGHT_SKIP_BROWSER_DOWNLOAD: 1
+    - run: npm run build
+    - run: npx playwright install --with-deps chromium
+    - run: xvfb-run --auto-servernum --server-args="-screen 0 1280x960x24" -- npm run test -- --project=${{ matrix.browser }}
```
Comment (yury-s) [inline .github/workflows/tests_secondary.yml:673]: There is no `matrix.browser` in this job

**Comment by mxschmitt** [comment]
The following tests are failing:

```
    [chromium] › library/browsercontext-credentials.spec.ts:20:3 › should fail without credentials ─
    [chromium] › library/browsercontext-credentials.spec.ts:31:3 › should work with setHTTPCredentials 
    [chromium] › library/browsercontext-credentials.spec.ts:102:3 › should fail with correct credentials and mismatching scheme 
    [chromium] › library/browsercontext-credentials.spec.ts:114:3 › should fail with correct credentials and mismatching hostname 
    [chromium] › library/browsercontext-credentials.spec.ts:128:3 › should fail with correct credentials and mismatching port 
    [chromium] › library/download.spec.ts:637:3 › should be able to download a inline PDF file via navigation 
    [chromium] › library/permissions.spec.ts:152:5 › permissions › should support clipboard read ───
    [chromium] › library/tracing.spec.ts:355:14 › should produce screencast frames fit ─────────────
    [chromium] › library/tracing.spec.ts:355:14 › should produce screencast frames crop ────────────
    [chromium] › library/tracing.spec.ts:355:14 › should produce screencast frames scale ───────────
    [chromium] › library/video.spec.ts:677:5 › screencast › should capture full viewport ───────────
    [chromium] › library/video.spec.ts:712:5 › screencast › should capture full viewport on hidpi ──
    [chromium] › library/video.spec.ts:749:5 › screencast › should work with video+trace ───────────
```

File: tests/library/video.spec.ts [hunk 2]
```diff
@@ -676,7 +676,7 @@ it.describe('screencast', () => {
 
   it('should capture full viewport', async ({ browserType, browserName, headless, isWindows }, testInfo) => {
     it.info().annotations.push({ type: 'issue', description: 'https://github.com/microsoft/playwright/issues/22411' });
-    it.fixme(browserName === 'chromium' && !headless, 'The square is not on the video');
+    it.fixme(browserName === 'chromium' && (!headless || !!process.env.PLAYWRIGHT_CHROMIUM_USE_HEADLESS_NEW), 'The square is not on the video');
```
Comment (yury-s) [inline tests/library/video.spec.ts:679]: Are they actually failing? The new mode is still headless and I'd not expect it to affect these tests.

File: tests/library/browsercontext-credentials.spec.ts [hunk 3]
```diff
@@ -17,8 +17,14 @@
 
 import { browserTest as it, expect } from '../config/browserTest';
 
-it('should fail without credentials', async ({ browser, server, browserName, headless }) => {
-  it.fail(browserName === 'chromium' && !headless);
+let isChromiumHeadedLike = false;
+
+it.beforeAll(async ({ browserName, headless }) => {
+  isChromiumHeadedLike = browserName === 'chromium' && (!headless || !!process.env.PLAYWRIGHT_CHROMIUM_USE_HEADLESS_NEW);
```
Comment (yury-s) [inline tests/library/browsercontext-credentials.spec.ts:23]: nit: make it a fixture?

**Review by yury-s — APPROVED** [review]

File: tests/library/video.spec.ts (hunk 2 above)
Comment (mxschmitt) [reply tests/library/video.spec.ts:679]: Yes they fail. Same backend.

**Pull request merged**